client/src/components/TimeoutDialog.vue

```diff
@@ -38,9 +38,13 @@ export default defineComponent({
       idleWarningTriggered.value = false;
     };
     const logout = async () => {
-      await djangoRest.logout();
-      // This will redirect to the login page
-      await djangoRest.login();
+      try {
+        // This may fail with a 401 if the user has already been logged out
+        await djangoRest.logout();
+      } finally {
+        // This will redirect to the login page
+        await djangoRest.login();
```

Comment:
If they're logged into the server, won't this just automatically re-log them into the client? If so, doesn't that defeat the purpose of the auto-logout?

Reply:
`djangoRest.logout` invokes a special API endpoint that explicitly terminates the server session. If that call succeeds, then the user is no longer logged in when `djangoRest.login` is called, so they get sent to the login page. If that call fails, then the user was not logged in, so they will still get sent to the login page.